File: RealState.Domian/Model/Propertys/Property.cs
using RealState.Domian.Model.Financial;

using RealState.Domian.Model.Users;

namespace RealState.Domian.Model.Propertys
{
    public class Property
    {
        public Guid Id { get; set; }
        public Typology? Type { get; set; }
        public Address? Location { get; set; }
        public int YearBuilt { get; set; }
        public double TotalArea { get; set; }
        public Rooms? TypedOfRooms { get; set; }
        public int Floors { get; set; }
        public Bathrooms? Bathrooms { get; set; }
        public Garage? Garage { get; set; }
        public bool Elevator { get; set; }
        public string? EnergyRating { get; set; }
        public Condition? ConditionOfUse { get; set; }
        public string? Description { get; set; }
        //public List<string?> Features { get; set; }
        public TypeOfUser? User { get; set; }
        public Price? Price { get; set; }
    }
}

File: RealState.Presentation/angular17/RealEstate/src/app/models/properties/property.ts
import { Address } from "../Addresses/address";
import { Condition } from "../Enums/condition";
import { Price } from "../Prices/price";
import { Typology } from "../typologeis/typology";
import { User } from "../Users/user";

export class Property {
    id: string;
    typology: Typology;
    location: Address;
    yearBuilt: number;
    totalArea: number;
    rooms: number;
    floors: number;
    bathrooms: number;
    garage: boolean;
    energyRating: string;
    condition: Condition;
    description: string;
    features: string;
    user: User;
    price: Price;

    constructor() {
        this.id = '';
        this.typology = new Typology();
        this.location = new Address();
        this.yearBuilt = 0;
        this.totalArea = 0;
        this.rooms = 0;
        this.floors = 0;
        this.bathrooms = 0;
        this.garage = false;
        this.energyRating = '';
        this.description = ''; 
        this.features = '';
        this.user = new User();
        this.price = new Price();
        this.condition = Condition.New;
    }
}
